app/src/main/java/net/bible/android/control/backup/BackupControl.kt

```diff
@@ -67,6 +76,55 @@ class BackupControl {
         }
     }
 
+    /** backup database to custom target (email, drive etc.)
+     */
+    fun backupDatabaseViaIntent(callingActivity: Activity) {
+        val fileName = CommonDatabaseHelper.DATABASE_NAME
+        internalDbBackupDir.mkdirs()
+        FileManager.copyFile(CommonDatabaseHelper.DATABASE_NAME, internalDbDir, internalDbBackupDir)
+
+		val subject = callingActivity.getString(R.string.backup_email_subject)
+		val message = callingActivity.getString(R.string.backup_email_message)
+        val f = File(internalDbBackupDir, fileName)
+        val uri = FileProvider.getUriForFile(callingActivity, BuildConfig.APPLICATION_ID + ".provider", f)
+		val email = Intent(Intent.ACTION_SEND_MULTIPLE).apply {
+            putExtra(Intent.EXTRA_SUBJECT, subject)
+            putExtra(Intent.EXTRA_TEXT, message)
+            putParcelableArrayListExtra(Intent.EXTRA_STREAM, arrayListOf(uri))
+            type = "text/*"
+        }
+		val chooserIntent = Intent.createChooser(email, "Send")
+        chooserIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+		callingActivity.startActivity(chooserIntent)
+    }
+
+    /** backup database from custom source
+     */
+    fun restoreDatabaseViaIntent(inputStream: InputStream) {
+        val fileName = CommonDatabaseHelper.DATABASE_NAME
+        internalDbBackupDir.mkdirs()
+        val f = File(internalDbBackupDir, fileName)
+        var ok = false
+        val header = inputStream.readExactNBytes(16)
+        if(String(header) == "SQLite format 3\u0000") {
+            val out = FileOutputStream(f)
+            out.write(header)
+            out.write(inputStream.readBytes())
+            out.close()
```

Comment:
I guess writing the file via input stream is always safe? Somehow I feel uneasy about that, but I guess it should not be an issue.

Reply:
Why not? How would you do this?
I'm reading from inputstream and writing into FileOutputStream.

Reply:
I was thinking of handling the Intent data as a file instead of input stream, but these are just my "feelings" per se. Don't have much experience on Android with either one of them.

Reply:
I think this is the only way you can do it. File content can come from any
source, not just from file system. Like Google Drive for example.

Reply:
Ok

Reply:
So this way you need to implement reading plan importing in you PR :)

Reply:
Already started working on it :)
I only use the inputStream anyway so haven't tried with file object yet.